src/components/Orders/OrdersHistoryItem.tsx

```diff
@@ -14,13 +14,21 @@ import { useQuery } from "react-query";
 import { useNavigate } from "react-router-dom";
 import Spinner from "../Spinner/Spinner";
 import { KEYQUERIES } from "../../common/constants";
-import { IOrdersQuery } from "../../common/interfaces";
+
+import { fetchOrders, selectAllOrders } from "../../store/order-slice";
+import { useDispatch, useSelector } from "react-redux";
+import { useEffect } from "react";
 
 const OrdersHistoryItem = () => {
-  const { isLoading, data, error } = useQuery<IOrdersQuery, Error>(
-    KEYQUERIES.orders,
-    getOrders
-  );
+  const orders = useSelector(selectAllOrders);
+  const error = useSelector((state: any) => state.orders.error);
+  const loading = useSelector((state: any) => state.orders.loading);
+  const dispatchAction = useDispatch();
```

Comment:
it's better to keep your selector functions somewhere in another file and export it for convinience / potential reusability etc
any 😢